underfs/swift/src/main/java/alluxio/underfs/swift/SwiftInputStream.java

```diff
@@ -83,6 +93,9 @@ public int read(byte[] b, int offset, int length) throws IOException {
     int read = mStream.read(b, offset, length);
     if (read != -1) {
       mPos += read;
+      if (mPos % getBlockSize() == 0) {
+        closeStream();
+      }
```

Comment:
@calvinjia 

scenario 1) read will do a partial read of 10 bytes and closeStream (sets mStream = null). next read operation will call openStream.

my understanding is that inputstream should always be called in a loop as the interface specifies that "An attempt is made to read as many as len bytes, but a smaller number may be read. The number of bytes actually read is returned as an integer."

scenario 2) 
case 1. a previous read() incremented the position to 20 => mStream will be set to 'null'; next read will call openStream. 
case 2. a skip() incremented the position to 20 => skip called openStream for range (20,40). read() does not return -1.